src/ImageStats/Histogram.cc

```diff
@@ -13,33 +13,52 @@
 
 using namespace carta;
 
-Histogram::Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data)
-    : _num_bins(num_bins), _bin_width((max_value - min_value) / num_bins), _min_val(min_value), _hist(num_bins, 0), _data(data) {}
+Histogram::Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data) :_data(data) {
+    _histogram.num_bins = num_bins;
```

Comment:
I'm wondering if we can just get rid of `num_bins`, since it's always equal to `histogram_bins.size()`

Reply:
I prefer to have the variable cached there even when the calling size is O(1), but that's a bit of a personal taste. I'd say it makes the code more readable and, in theory, negligible faster. I'd rather change its type to `unsigned int`.